spyder/app/mainwindow.py

```diff
@@ -2240,6 +2241,17 @@ def __update_fullscreen_action(self):
         self.fullscreen_action.setIcon(icon)
 
     @Slot()
+    def undock_editor(self):
+        """Open a new window instance of the Editor instead of undocking it."""
+        if self.editor.dockwidget.isFloating() and not self.editor.undocked:
+            self.editor.dockwidget.setVisible(False)
+            self.editor.create_new_window()
+            self.editor.toggle_view_action.setChecked(False)
+            self.editor.dockwidget.setFloating(False)
+        self.editor.undocked = False
+        self.editor.get_current_editorstack().new_window = False

```

Comment:
@dalthviz, does this method need to be here in the main window? As far as I can see, it doesn't uses any attribute/method of the window, so I think we can safely move it to the the Editor plugin file.